tests/benchdnn/graph/input_displacer.cpp

```diff
@@ -136,6 +136,120 @@ partition_data_displacer_t::partition_data_displacer_t(
+                // unlike for implicit causal mask, it's required to update the
+                // condition to always take qk values instead of a special one.
+                //
+                // Checking for data type to make sure that in case of two user
+                // inputs, the condition one will be updated. For implicit
+                // causal mask, the condition would have a parent and a check
+                // for `causal_mask_lt` being non-empty will fail.
+                if (causal_mask_lt
+                        && aop_in_lt->get_data_type()
+                                != logical_tensor::data_type::boolean)
+                    continue;
+
+                causal_mask_lt = aop_in_lt;
+                offset = i;
+            }
+            // No suitable tensor/subgraph for a mask displacement.
+            if (!causal_mask_lt) break;
+
+            filling_type_t filling_type = filling_type_t::undef;
+            if (aop.kind_ == "Add") {
+                const auto ndims = causal_mask_lt->shape_.size();
+                if (ndims < 2) {
+                    BENCHDNN_PRINT(7, "%s\n",
+                            "[DISPLACE]: Causal mask ndims is less than 2");
+                    break;
+                }
+
+                const auto M = causal_mask_lt->shape_[ndims - 2];
+                if (M == 1) {
+                    // This is a padding mask case, when padded tokens should
+                    // be removed from the final computations. In case of
+                    // benchdnn, there's no such thing as padding as all tokens
+                    // are computed. To avoid numerical instabilities, a zero
+                    // mask can be applied without compromising validation
+                    // capabilities.
+                    filling_type = filling_type_t::zero;
+                } else {
+                    // This is a look-ahead (or causal) mask case, when future
+                    // tokens (row < col) are set to infinity to remove all
+                    // connections of current tokens to unissued ones.
+                    filling_type = filling_type_t::causal_mask;
+                }
+            } else if (aop.kind_ == "Select") {
+                if (qk_data_offset == SIZE_MAX) {
+                    // Implicit causal mask case.
+                    filling_type = filling_type_t::minus_infinity;
+                } else {
+                    // Padding mask.
+                    assert(qk_data_offset == 1 || qk_data_offset == 2);
+                    // Fill condition depending on qk values tensor to use only
+                    // its values, which is equivalent of not using a mask.
+                    if (qk_data_offset == 1) {
+                        filling_type = filling_type_t::one;
+                    } else if (qk_data_offset == 2) {
+                        filling_type = filling_type_t::zero;
```

Comment:
Could you please elaborate why the filling type is one or zero regarding qk_data_offset = 1 / 2?

Reply:
Select takes the value from src_0 or src_1 depending on the value in cond. The desired value is always the one coming from qk_matmul. Depending on its position in the op, condition should use `1` to pick those values from src_0, or `0` to pick it from src_1.

Reply:
Sorry, I thought it was filling the mask instead of the condition. If it's filling the condition, then make sense 👍